xclim/testing/tests/test_indicators.py

```diff
@@ -461,7 +461,11 @@ def test_all_parameters_understood(official_indicators):
             if param["kind"] == InputKind.OTHER_PARAMETER:
                 problems.add((identifier, name))
     # this one we are ok with.
-    if problems - {("COOL_NIGHT_INDEX", "lat")}:
+    if problems - {
+        ("COOL_NIGHT_INDEX", "lat"),
+        ("DRYNESS_INDEX", "lat"),
+        ("DRYNESS_INDEX", "wo"),
+    }:
```

Comment:
`wo`  is a `Quantified`, so should it not avoid the `InputKind` error? 

Also, this is less important and I'm just thinking outloud:  `lat` is not a `Quantified` so that it can be a `None` type?  Perhaps  `Quantified` ought to allow `None`? Another strategy could be to replace the use of `None` with an empty string for `lat`, but perhaps it's not  very clean and might as well keep this exception above.

Reply:
I think what complicates things for `lat` is that it can optionally be a `str` instance as well (in the case where we supply a point with no `lat` information easily found, like station obs data). An empty string might raise an exception with the way CNI and DI are currently written.

`wo` here is not `optional` and must have a `units` value. It's hardcoded in the indice. Not sure if I follow why we would allow `None` here, unless there are other cases ?

Reply:
Ok, thanks, clear up things I was confused with in another PR!

Reply:
`Quantified` inputs must all be polymorphic so that in the example below `A_conv` can be used for arithmetic with xarray DataArrays. Which  means is must either be a string (quantity with units, `A_conv` will be a float) or a `DataArray` with units (`A_conv` will be a DataArray).

```python3

A : Quantified = <something>

A_conv = convert_units_to(A, 'units')
```

Sadly this does not allow `None`, we are left with this test-skipping for `lat`...

But indeed, `wo` shouldn't have to appear here!

Reply:
> But indeed, `wo` shouldn't have to appear here!

Interesting. I have test failures otherwise. Bug?

Reply:
For the empty string for lat, I was indeed thinking to replace the condition that checks if `lat` is `None` and check if `lat` is an empty string instead, so it would mean some rewriting. I don't think this approach is used elsewhere in XClim though, that was just to explore possibilities to avoid the exception. But perhaps the exception is a lesser evil